src/qutip_tensornetwork/core/data/tensor_train/tensor_train.py

```diff
@@ -234,6 +234,119 @@ def _to_tt_format(self):
 
         self._nodes = set(nodes)
 
+    def truncate(self, bond_dimension=None, max_truncation_err=None):
+        """Truncate in-place the bond dimension of the tensor train according
+        to ``bond_dimension``. This is done from left to right and once
+        finished the network is right normalized.
+
+        Notes
+        -----
+        The truncation method consists on performing a svd decomposition from
+        left to right and truncating each nodes svd by the given parameters.
+
+        Parameters
+        ----------
+        bond_dimension: List of int or int
+            List of integers that define, from left to right, the target bond
+            dimension. If a single integer is provided is understood as the
+            same bond dimension for all bonds. None is understood as
+            "infinite" bond dimension and hence no truncation is done.
+
+        max_truncation_err: List of float or float
```

Comment:
We should:
- method that `max_truncation_err` can also be `None` (what that means)
- explain a bit more how the bond_dimension and max_truncation_err interact (i.e. bond_dimension takes precedence)